lib/checks/landmarks/landmark-is-unique-after.js

```diff
@@ -0,0 +1,20 @@
+const uniqueLandmarks = [];
+return results.filter(currentResult => {
```

Comment:
This solution isn't going to work with scoped runs. This check is assuming that the scope of the run is always the entire page, rather than some element or section on the page. Take this example:

```html
<nav aria-label="foo" id="foo"></nav>
<nav aria-label="foo" id="bar"></nav>
<script> axe.run('#foo') </script>
```

Because `#bar` is out of scope when running axe, it will not be in the `results` array in the after method, and so `#foo` won't be seen as a duplicate. The way to deal with this is to have the `evaluate` method look for duplicates in the page. Have a look at the duplicate-id rule for an example. That also avoids the need of updating relatedNodes in the after method, which while it does work, isn't how this is supposed to work.

General rule of thumb is that when you test the page as a whole, the total number of issues should be the same as when you test separate pieces of the page and concatenate the results.

Reply:
After some extensive discussions, I'm going to withdraw this comment. Dylan and I can't come up with a way to have this rule work across frames, and with limited scope. At least not one that doesn't require substantial changes to the engine.

We think testing landmarks across frames is more important than comparing landmarks outside the scope of the run. So lets go with the proposed solution.

There are other comments to resolve though, so please pick those up, and leave this one for what it is.

Reply:
Resolving as such.